### PR TITLE
[OSDEV-1332] Moderation. Implement PATCH api/v1/moderation-events/{moderation_id}

### PR DESCRIPTION
**Pull request by roman-stolar**
[OSDEV-1332](https://opensupplyhub.atlassian.net/browse/OSDEV-1332) Moderation. Implement PATCH api/v1/moderation-events/{moderation_id}

- Create new PATCH request for api/v1/moderation-events/{moderation_id}
- Updated error handling and provided 200 response data

[OSDEV-1332]: https://opensupplyhub.atlassian.net/browse/OSDEV-1332?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by roman-stolar** [comment]
@coderabbit Please review this PR.

**Review by mazursasha1990 — COMMENTED** [review]
Have several questions/comments:

1. Do you really want to merge these changes into `OSDEV-1346-implement-get-moderation-events` not `main`?
2. Please add Release Notes.
3. In my opinion comments to code are redundant.

File: src/django/api/serializers/v1/moderation_event_update_serializer.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+            raise ValidationError({
+                "field": "status",
+                "message": "This field is required."
+            })
+
+        # Invoke validate_status to enforce custom validation
+        self.validate_status(status)
+        return super().to_internal_value(data)
+
+    def validate_status(self, value):
```
[the remark this answers is not in the document]
Comment (roman-stolar) [reply src/django/api/serializers/v1/moderation_event_update_serializer.py:52]: Yep, good point.

**Comment by roman-stolar** [comment]
@mazursasha1990 
1. It will be replaced to `main` branch when `OSDEV-1346-implement-get-moderation-events` will be merged as it depend on it.
2. Will add.
3. I see, I can removed the comments.

File: src/django/api/serializers/v1/moderation_event_update_serializer.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (roman-stolar) [reply src/django/api/serializers/v1/moderation_event_update_serializer.py:52]: Done.

**Comment by roman-stolar** [comment]
The [PR](https://github.com/opensupplyhub/open-supply-hub-api-docs/pull/21) with updated API specification.

**Review by mazursasha1990 — APPROVED** [review]
LGTM

File: src/django/api/views/v1/moderation_events.py [hunk 2]
```diff
@@ -45,10 +50,50 @@ def list(self, request):
...
             OpenSearchIndexNames.MODERATION_EVENTS_INDEX,
             query_body
         )
         return Response(response)
+
+    @handle_errors_decorator
+    def patch(self, request, moderation_id):
+        if not (request.user.is_superuser or request.user.is_staff):
+            raise PermissionDenied(
+                detail="Only Moderator could perform this action."
```
[the remark this answers is not in the document]
Comment (roman-stolar) [reply src/django/api/views/v1/moderation_events.py]: Done

**Review by mazursasha1990 — DISMISSED** [review]
LGTM

**Review by mazursasha1990 — APPROVED** [review]
LGTM

**Review by mazursasha1990 — APPROVED** [review]
LGTM

**Review by VadimKovalenkoSNF — APPROVED** [review]
Approved.

**Pull request merged**